test/python/circuit/test_circuit_operations.py

```diff
@@ -1002,6 +1002,31 @@ def test_reverse(self):
 
         self.assertEqual(qc.reverse_ops(), expected)
 
+    def test_reverse_with_standlone_vars(self):
+        """Test that instruction-reversing works in the presence of stand-alone variables."""
+        a = expr.Var.new("a", types.Bool())
+        b = expr.Var.new("b", types.Uint(8))
+        c = expr.Var.new("c", types.Uint(8))
+
+        qc = QuantumCircuit(2, inputs=[a])
+        qc.add_var(b, 12)
+        qc.h(0)
+        qc.cx(0, 1)
+        with qc.if_test(a):
+            # We don't really comment on what should happen to control-flow operations in this
+            # method, and Sabre doesn't care (nor use this method, in the fast paths), so this is
+            # deliberately using a body of length 1 (a single `Store`).
```

Comment:
I'm a bit confused where is the sabre discussion coming from in this test? This is validating that `reverse_ops()` works correctly with variables.

Reply:
`QuantumCircuit.reverse_ops` is only (meaningfully) used by the slow-path `SabreLayout` using an explicit routing pass, but thinking more, the idea of "Sabre doesn't care what happens within control-flow ops" only applies the same way to the fast path, so the Sabre bit is irrelevant.  Changed in 4dedd6cd38.